app/controllers/comments_controller.rb

```diff
@@ -342,4 +350,18 @@ def comment_rate_limited
     end
     false
   end
+
+  def check_if_thread_is_private
+    if @comment_thread&.is_private
+      flash[:danger] = 'This action is not permitted.'
+      redirect_to comment_thread_path(@comment_thread.id)
+    end
+  end
+
+  def check_if_not_mod_and_thread_is_private
+    if !current_user.is_moderator
+      check_if_thread_is_private
```

Comment:
A redirect from a `before_action` method prevents the action it precedes from running. Does this still apply if it redirects indirectly (by calling another method that redirects)?

If so, this is fine. Otherwise, the code from `check_if_thread_is_private` may need to be inlined into `check_if_not_mod_and_thread_is_private`.

Reply:
I have confirmed by experiment that the filter chain is halted when a `before_action` renders or redirects, even if that render or redirect is done indirectly by calling another method. So nothing needs to be changed here.